include/dkm_utils.hpp

```diff
@@ -124,4 +125,44 @@ std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>> get_best_means(
 	return best_means;
 }
 
+
+
+
+    /**
+	 * Return the best clustering obtained from a given number of k-means
+	 * calculations.
+	 *
+	 * @param points  Sequence of points to be clustered.
+	 * @param k		  Number of clusters
+	 * @param n_init  Number of times a k-means clustering will be calculated.
+	 *
+	 * @return Clustering with the lowest inertia.
+	 */
+
+
+	template <typename T, size_t N>
+	std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>> get_best_means_parallel(
+	const std::vector<std::array<T, N>>& points, const clustering_parameters<T>& parameters, uint32_t n_init = 10) {
+
+		// seed generator once 
+		std::random_device rand_device;
+		uint64_t seed = parameters.has_random_seed() ? parameters.get_random_seed() : rand_device();
+		std::linear_congruential_engine<uint64_t, 6364136223846793005, 1442695040888963407, UINT64_MAX> rand_engine(seed);
```

Comment:
please use the `CustomGenerator` that was declared at the included `dkm_parallel.hpp` file